pkg/auth/providerrefresh/refresher.go

```diff
@@ -206,6 +206,10 @@ func (r *refresher) refreshAttributes(attribs *v3.UserAttribute) (*v3.UserAttrib
 			secret := ""
 			if providers.ProvidersWithSecrets[providerName] {
 				secret, err = r.tokenMGR.GetSecret(user.Name, providerName, loginTokens[providerName])
+				if apierrors.IsNotFound(err) {
+					// There is no secret so we can't refresh, just continue to the next attribute
+					return attribs, nil
```

Comment:
I think this should be continue instead of a return.

Reply:
This is to keep the same behavior that is done today.  Currently it will error out and never continue, but causes the handler to infinitely retry.  This way we don't retry anymore.  I'm concerned with continuing as I don't know what the correct behavior is in this situation as we haven't had this before.  Do we keep existing principals or delete them?  I think it's safer to just fix the performance issue and not address the logic issue.